platforms/ide/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApiSpecification.groovy

```diff
@@ -306,8 +306,10 @@ abstract class ToolingApiSpecification extends Specification implements KotlinDs
         assert failureOutput.contains("BUILD FAILED")
     }
 
-    void assertHasConfigureSuccessfulLogging() {
-        assertHasNoDeprecationWarnings()
+    void assertHasConfigureSuccessfulLogging(boolean hasDeprecations = false) {
+        if (!hasDeprecations) {
+            assertHasNoDeprecationWarnings()
+        }
```

Comment:
We need to avoid asserting when we're expecting a deprecation warning, I figured the invisible default param wouldn't really even be noticed here.

I can split this into 2 `assertHasConfigureSuccessfulLogging()` and `assertHasConfigureSuccessfulLoggingWithDeprecations()` if you think that's better?

Reply:
Lets consider how something like this works for a regular integ test -- where we fail if any deprecations are printed that are not explicitly listed before running the build. I would imagine we'd want something similar here. 

Currently we do:
```
assertHasConfigureSuccessfulLogging() // Asserts there are no deprecation warnings
result.assertOutputContains("Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.")
```


Instead, we should do something like below, like we do for integ/smoke tests.

```
expectDeprecation("Resolution of the configuration :a:compileClasspath was attempted from a context different than the project context.")
assertHasConfigureSuccessfulLogging() // Asserts there are no deprecation warnings other than those listed prior to calling this method
```

I think we can implement a very rudimentary version of this in `ToolingApiSpecification` by updating (and renaming) `assertHasNoDeprecationWarnings` to ensure only explicitly listed deprecations are present. We could even do `startsWith` checks to make it easier to do comparisons -- so we don't need to check for the URL and handle the changing version.


Alternatively, we can keep it simple and do something like 

```
expectDeprecationEmitted()
assertHasConfigureSuccessfulLogging()
```